tests/conftest.py

```diff
@@ -73,6 +73,10 @@ def base_app():
         SECURITY_PASSWORD_SALT='TEST_SECURITY_PASSWORD_SALT',
         SECRET_KEY='TEST_SECRET_KEY',
     )
+    # update the application with the configuration provided by the test
```

Comment:
In this case I am not sure that the tests are bigger. Only the tests needing a specific application configuration are parametrized. As we set the configuration before initializing the extensions, we would need to move all this init in each test, thus making them even bigger and duplicating code.
It also enables easy testing with different configurations.
However it's true that the readability is far from being perfect.

Or we can also suppose that the options can be set after init time (in this case it would work), but I would prefer to test the code the same way we recommend to use it.